scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,580 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Exit on error
+set -e
+
+
+# Global variables
+TIMESTAMP="$(date +%F_%T)"
+TMPFOLDER="$(mktemp -d -p . "kubestellar-snapshot-XXXX")"
+trap 'rm -rf -- "$TMPFOLDER"' EXIT
+OUTPUT_FOLDER="$TMPFOLDER/kubestellar-snapshot"
+
+
+# Script info
+SCRIPT_NAME="KubeStellar Snapshot"
+SCRIPT_VERSION="0.2.0"
+
+
+# Colors
+COLOR_NONE="\033[0m"
+COLOR_RED="\033[1;31m"
+COLOR_GREEN="\033[1;32m"
+COLOR_BLUE="\033[94m"
+COLOR_YELLOW="\033[1;33m"
+COLOR_PURPLE="\033[1;35m"
+COLOR_WARNING="${COLOR_RED}"
+COLOR_ERROR="${COLOR_RED}"
+COLOR_STATUS_TRUE="${COLOR_GREEN}"
+COLOR_STATUS_FALSE="${COLOR_RED}"
+COLOR_INFO="${COLOR_BLUE}"
+COLOR_TITLE="${COLOR_YELLOW}"
+COLOR_YAML="${COLOR_PURPLE}"
+
+
+# Command line arguments
+arg_kubeconfig=""
+arg_context=""
+arg_logs=false
+arg_yaml=false
+arg_verbose=false
+
+
+# Display the command line help
+display_help() {
+  cat << EOF
+Usage: $0 [options]
+
+Options:
+--kubeconfig|-K <filename> use the specified kubeconfig to find KubeStellar Helm chart
+--context|-C <name>        use the specified context to find KubeStellar Helm chart
+--logs|-L                  save the logs of the pods
+--yaml|-Y                  save the YAML of the resources
+--verbose|-V               output extra information
+--version|-v               print out the script version
+--help|-h                  show this information
+-X                         enable verbose execution for debugging
+
+Note 1: This script expects that KubeStellar was installed with its Helm chart and
+        to find the latter in the host or provided kubeconfig(s) and context(s).
```

Comment:
s/to/tries to/
s/in the host or provided kubeconfig(s) and context(s)/in a context of the relevant kubeconfig/